yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/call/aml/RemoteAmlService.java

```diff
@@ -1,54 +1,57 @@
 package com.yoti.api.client.spi.remote.call.aml;
 
+import static java.net.HttpURLConnection.HTTP_BAD_REQUEST;
+import static java.net.HttpURLConnection.HTTP_INTERNAL_ERROR;
+import static java.net.HttpURLConnection.HTTP_UNAUTHORIZED;
+
 import static com.yoti.api.client.spi.remote.call.HttpMethod.HTTP_POST;
-import static com.yoti.api.client.spi.remote.call.YotiConstants.CONTENT_TYPE;
-import static com.yoti.api.client.spi.remote.call.YotiConstants.CONTENT_TYPE_JSON;
 import static com.yoti.api.client.spi.remote.call.YotiConstants.DEFAULT_CHARSET;
 import static com.yoti.api.client.spi.remote.call.YotiConstants.DEFAULT_YOTI_API_URL;
-import static com.yoti.api.client.spi.remote.call.YotiConstants.DIGEST_HEADER;
-import static com.yoti.api.client.spi.remote.call.YotiConstants.JAVA;
 import static com.yoti.api.client.spi.remote.call.YotiConstants.PROPERTY_YOTI_API_URL;
-import static com.yoti.api.client.spi.remote.call.YotiConstants.SDK_VERSION;
-import static com.yoti.api.client.spi.remote.call.YotiConstants.YOTI_SDK_HEADER;
-import static com.yoti.api.client.spi.remote.call.YotiConstants.YOTI_SDK_VERSION_HEADER;
 import static com.yoti.api.client.spi.remote.util.Validation.notNull;
-import static java.net.HttpURLConnection.HTTP_BAD_REQUEST;
-import static java.net.HttpURLConnection.HTTP_INTERNAL_ERROR;
-import static java.net.HttpURLConnection.HTTP_UNAUTHORIZED;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.util.Map;
+
 import com.yoti.api.client.AmlException;
 import com.yoti.api.client.aml.AmlProfile;
 import com.yoti.api.client.spi.remote.call.JsonResourceFetcher;
 import com.yoti.api.client.spi.remote.call.ResourceException;
 import com.yoti.api.client.spi.remote.call.ResourceFetcher;
 import com.yoti.api.client.spi.remote.call.UrlConnector;
+import com.yoti.api.client.spi.remote.call.factory.HeadersFactory;
 import com.yoti.api.client.spi.remote.call.factory.PathFactory;
 import com.yoti.api.client.spi.remote.call.factory.SignedMessageFactory;
 
-import java.io.IOException;
-import java.security.GeneralSecurityException;
-import java.security.KeyPair;
-import java.util.HashMap;
-import java.util.Map;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
 public class RemoteAmlService {
 
     private final PathFactory pathFactory;
+    private final HeadersFactory headersFactory;
     private final ObjectMapper objectMapper;
     private final SignedMessageFactory signedMessageFactory;
     private final ResourceFetcher resourceFetcher;
     private final String apiUrl;
 
     public static RemoteAmlService newInstance() {
-        return new RemoteAmlService(JsonResourceFetcher.newInstance(), new PathFactory(), new ObjectMapper(), SignedMessageFactory.newInstance());
+        return new RemoteAmlService(
```

Comment:
That's Dependency Injection.
https://en.wikipedia.org/wiki/Dependency_injection
https://www.journaldev.com/2394/java-dependency-injection-design-pattern-example-tutorial

If we stick with the approach of using composition (and we will) then adding new functionality to the SDKs will result in creating new components, and those components will have to be injected in to the clients we expose.

This SDK does not have any DI framework, because we've always tried to minimise the list of dependencies it ships with.  As a consequence, the component lifecycle is turning out a little messy.  It would be a good thing to fix it (or reconsider our position on DI frameworks), but that would be a v3 change.